saleor/core/utils/tests/test_url.py

```diff
@@ -51,3 +54,48 @@ def test_get_default_storage_root_url(site_settings):
 )
 def test_sanitize_url_for_logging(url, expected):
     assert sanitize_url_for_logging(url) == expected
+
+
+def _refresh_storages():
+    """
+    pytest, as tests runner doesn't reload Django-based settings, so we need to force it to refresh them,
+    when we want to test how settings will behave based on different environment variables.
+    """
+    importlib.reload(sys.modules["saleor.settings"])
+    importlib.reload(sys.modules["saleor.tests.settings"])
+    del settings.STORAGES
+    settings._setup()
+    storages._backends = None
+    storages._storages = {}
+    del storages.backends
+    default_storage._setup()
```

Comment:
We shouldn't do that, it will affect other tests. If we want to test a code path, we should extract that piece of code to a function.

Reply:
I agree it looks terrible, AFAIK whole 13k tests passes (I tested that locally), but I also agree, that there's no justifiable value provided be these tests to maintain this refresh logic (and ensure it's not going to affect other tests).

> If we want to test a code path, we should extract that piece of code to a function.

Here I would defer making that decision to Django 5, as it looks there is major rewrite in a way Django treats `STORAGES`.